functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobExecutionSpec.groovy

```diff
@@ -1900,6 +1900,122 @@ class JobExecutionSpec extends BaseContainer {
+                ]
+        ]
+
+        def responseProject = createSampleProject(projectJsonMap)
+        assert responseProject.successful
+
+        String childJobId = "a472767a-6575-46f1-8ab4-6cd2b289f1f3"
+        def testXml = """
+            <joblist>
+               <job>
+                  <name>Child</name>
+                  <group>api-test/job-run</group>
+                  <uuid>${childJobId}</uuid>
+                  <description></description>
+                  <loglevel>INFO</loglevel>
+                  <context>
+                      <project>${projectName}</project>
+                  </context>
+                  <dispatch>
+                    <threadcount>1</threadcount>
+                    <keepgoing>true</keepgoing>
+                  </dispatch>
+                  <sequence>
+                    <command>
+                    <exec>echo 1</exec>
+                    </command>
+                  </sequence>
+               </job>
+               <job>
+                  <name>Parent</name>
+                  <group>api-test/job-run</group>
+                  <uuid>14e09b60-436c-42b0-b9dc-6308f1fbd5cc</uuid>
+                  <description/>
+                  <executionEnabled>true</executionEnabled>
+                  <loglevel>INFO</loglevel>
+                   <context>
+                      <project>${projectName}</project>
+                  </context>
+                  <nodeFilterEditable>false</nodeFilterEditable>
+                  <sequence keepgoing=\"false\" strategy=\"parallel\">
+                     <command>
+                         <jobref name=\"Child\">
+                           <uuid>${childJobId}</uuid>
+                         </jobref>
+                     </command>                     
+                  </sequence>
+               </job>
+            </joblist>
+            """
+
+        def created = JobUtils.createJob(projectName, testXml, client)
+        assert created.successful
+
+        when: "Job and referenced job created"
+        CreateJobResponse jobCreatedResponse = MAPPER.readValue(
+                created.body().string(),
+                CreateJobResponse.class
+        )
+        def childJob = jobCreatedResponse.succeeded[0]?.id
+        def parentJob = jobCreatedResponse.succeeded[1]?.id
+
+        then:
+        childJob != null
+        parentJob != null
+
+        when: "run job"
+        def jobRun = JobUtils.executeJob(parentJob, client)
+        assert jobRun.successful
+
+        Execution exec = MAPPER.readValue(jobRun.body().string(), Execution.class)
+
+        Execution jobExecutionStatus = JobUtils.waitForExecution(
+                ExecutionStatus.SUCCEEDED.state,
+                exec.id as String,
+                client,
+                WaitingTime.EXCESSIVE)
+
+        then:
+        jobExecutionStatus.status == ExecutionStatus.SUCCEEDED.state
+
+        when: "get executions"
+        sleep 5000
```

Comment:
Why is this sleep necessary?

Reply:
I removed this sleep. I dont think it will be needed since it`s already calling `waitForExecution` method